### DISCUSSION TITLE
Doubts with usage

### DISCUSSION THREAD

**Discussion by ergonzal22** [gavv/httpexpect#127] [Q&A] [answered]
Hi,

I have some questions about the usage of this framework. Some features I can't find in the docs so I'm curious if there is a way to do them:

- Is there a way to stop running some of the tests if one of them fails and maybe go to a "clean" method. I can't find a way to do an `if` around a `request` and do some action if the test fails or not.
- Related to that. Is there a way to wait for a test to pass with a limited number of retries? I know I can do this by making requests using the http Go library, but I'm curious if there is a way of doing that with httpexpect

Thank you very much,
Regards,
Ernesto Gonzalez

**Comment by fpeterschmitt** [comment]
To stop a test from continuing you must instantiate like this:

```
	cfg := httpexpect.Config{
		BaseURL:  "yourBaseURL",
		Reporter: httpexpect.NewRequireReporter(t),
	}
	e := httpexpect.WithConfig(cfg)
```

**Comment by gavv** [comment, marked as answer]
Hi, thanks for asking.

> Is there a way to stop running some of the tests if one of them fails and maybe go to a "clean" method. I can't find a way to do an if around a request and do some action if the test fails or not.

As already answered, using httpexpect.NewRequireReporter (or your custom reporter) is the right way to do it.

> Related to that. Is there a way to wait for a test to pass with a limited number of retries? I know I can do this by making requests using the http Go library, but I'm curious if there is a way of doing that with httpexpect

There is no way to do it out of the box. There was a PR but it was not finished: #73. I've created an issue: #85.
